examples/SmartShop/contracts/SellerContract.aes

```diff
@@ -0,0 +1,36 @@
+contract Seller =
+  record state = {seller_key : address, buyer_contract: address, status : string, price : int}
+  
+  public stateful function init(bc : address, price : int) : state =
+    {seller_key = Call.caller, buyer_contract = bc, status = "undefined", price = price }
+    
+  public stateful function send_item() : bool =
+    if(check_money())
+      put(state{status = "sent_to_transport_courier"})
+      true
+    else
+      abort("Not enough money")
+      
+  public stateful function received_item() : bool =
+    if(Call.caller == state.buyer_contract)
+      put(state{status = "delevered"})
+      Chain.spend(state.seller_key, state.price)
+      true
+    else
+      abort("Only the buyer can use this function")
+      
+  public function key_balance() : int =
+    Chain.balance(Call.caller)
+    
+  public function seller_contract_balance() : int =
+    Contract.balance
+    
+  public function check_item_status() =
+    state.status
+
+  private function check_money() : bool =
+    if(Contract.balance >= state.price)
```

Comment:
can you add some kind of map that store every deposit.

Reply:
We would prefer to keep it simple as it is.